addons/crm/tests/test_crm_lead.py

```diff
@@ -423,6 +424,20 @@ def test_crm_lead_stages(self):
         self.assertEqual(lead.probability, 100.0)
         self.assertEqual(lead.stage_id, self.stage_gen_won)  # generic won stage has lower sequence than team won stage
 
+    @users('user_sales_manager')
+    def test_crm_last_stage_update_time(self):
+        """ Test date_last_stage_update update only when stage will change not user_id """
+        lead = self.lead_1.with_user(self.env.user)
+        current_time = datetime.now().replace(microsecond=0)
+        self.assertEqual(lead.date_last_stage_update, current_time)
```

Comment:
This seems like it could introduce non-deterministic behavior in some edge cases (0.999 != 1 kind of deal)

We should rather use freeze_time around this block to ensure the datetime will always be the same.
Then get rid of the `time.sleep()` (and the import) and just use a different freeze_time around that. Also makes the test much faster.

Reply:
Hello @thiryr ,
I made the changes, please have a look at it.
Thanks!